--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -2331,6 +2331,10 @@ config("chromium_code") {
         # Warn on GNU extensions.
         "-Wgnu",
 
+        # Ignore -Wgnu-line-marker, because clang's preprocessor itself
+        # generates these, and proceeeds to warn about its own output.
+        "-Wno-gnu-line-marker",
+
         # TODO(crbug.com/357081796): Try to enable these.
         "-Wno-gnu-anonymous-struct",
         "-Wno-gnu-conditional-omitted-operand",
