--- a/retrieve_and_unpack_resource.sh
+++ b/retrieve_and_unpack_resource.sh
@@ -109,5 +109,7 @@ if $retrieve_arch_specific; then
     _llvm_dir="$_src_dir/third_party/llvm-build/Release+Asserts"
     _llvm_bin_dir="$_llvm_dir/bin"
 
+    tar -x -C "$_llvm_dir" --strip-components=1 -f "$_root_dir/../clang+llvm-21.1.5-arm64-apple-darwin21.0.tar.xz"
+
     ln -s "$_llvm_bin_dir/llvm-install-name-tool" "$_llvm_bin_dir/install_name_tool"
 fi
