--- a/downloads-arm64.ini
+++ b/downloads-arm64.ini
@@ -1,14 +1,5 @@
 # arm64 (Apple Silicon) specific dependencies not included in the main Chromium source archive
 
-# Pre-built LLVM toolchain for convenience
-[llvm]
-version = bd809ffb4b5f277a661509fbbbf9ea893a545ab0
-url = https://github.com/dumbmoron/llvm-macos-buildbot/releases/download/%(version)s-arm64/clang+llvm-%(version)s-arm64-apple-darwin21.0.tar.xz
-download_filename = clang+llvm-%(version)s-arm64-apple-darwin21.0.tar.xz
-strip_leading_dirs = clang+llvm-%(version)s-arm64-apple-darwin21.0
-sha512 = 64ebdb038a8e0444e50032951775e75e40ae7a8cc8cdaa3d5148060666419ffee0d4579e9a400f5f945bd4e7e2540ff65e650df54eaada4f17eea75de3a4e0df
-output_path = third_party/llvm-build/Release+Asserts
-
 [nodejs]
 version = 22.11.0
 url = https://nodejs.org/dist/v%(version)s/node-v%(version)s-darwin-arm64.tar.xz
--- a/downloads-x86-64.ini
+++ b/downloads-x86-64.ini
@@ -1,14 +1,5 @@
 # x86-64 (Intel) specific dependencies not included in the main Chromium source archive
 
-# Pre-built LLVM toolchain for convenience
-[llvm]
-version = bd809ffb4b5f277a661509fbbbf9ea893a545ab0
-url = https://github.com/dumbmoron/llvm-macos-buildbot/releases/download/%(version)s-x86-64/clang+llvm-%(version)s-x86-64-apple-darwin21.0.tar.xz
-download_filename = clang+llvm-%(version)s-x86-64-apple-darwin21.0.tar.xz
-strip_leading_dirs = clang+llvm-%(version)s-x86-64-apple-darwin21.0
-sha512 = 979e514bb1741e1871128ed4dcdf600d97e48aaa746e337f246b1dd9f30a0f88833c02f31cefe507784108eeee854b3e611ad94bf4a22212dbd5fb7fe2875b11
-output_path = third_party/llvm-build/Release+Asserts
-
 [nodejs]
 version = 22.11.0
 url = https://nodejs.org/dist/v%(version)s/node-v%(version)s-darwin-x64.tar.xz
--- a/retrieve_and_unpack_resource.sh
+++ b/retrieve_and_unpack_resource.sh
@@ -109,5 +109,7 @@ if $retrieve_arch_specific; then
     _llvm_dir="$_src_dir/third_party/llvm-build/Release+Asserts"
     _llvm_bin_dir="$_llvm_dir/bin"
 
+    tar -x -C "$_llvm_dir" --strip-components=1 -f "$_root_dir/../clang+llvm-21.1.0-arm64-apple-darwin21.0.tar.xz"
+
     ln -s "$_llvm_bin_dir/llvm-install-name-tool" "$_llvm_bin_dir/install_name_tool"
 fi
